openstudiocore/src/model/AirflowNetworkDuctViewFactors.cpp

```diff
@@ -0,0 +1,309 @@
+  }
+
+  double AirflowNetworkDuctViewFactors_Impl::ductSurfaceEmittance() const
+  {
+    boost::optional<double> value = getDouble(OS_AirflowNetworkDuctViewFactorsFields::DuctSurfaceEmittance,true);
+    OS_ASSERT(value);
+    return value.get();
+  }
+
+  bool AirflowNetworkDuctViewFactors_Impl::isDuctSurfaceEmittanceDefaulted() const
+  {
+    return isEmpty(OS_AirflowNetworkDuctViewFactorsFields::DuctSurfaceEmittance);
+  }
+
+  bool AirflowNetworkDuctViewFactors_Impl::setLinkage(const AirflowNetworkLinkage& linkage)
+  {
+    bool result = setPointer(OS_AirflowNetworkDuctViewFactorsFields::LinkageName, linkage.handle());
+    return result;
+  }
+
+  bool AirflowNetworkDuctViewFactors_Impl::setDuctSurfaceExposureFraction(double ductSurfaceExposureFraction)
+  {
+    bool result = setDouble(OS_AirflowNetworkDuctViewFactorsFields::DuctSurfaceExposureFraction, ductSurfaceExposureFraction);
+    return result;
+  }
+
+  void AirflowNetworkDuctViewFactors_Impl::resetDuctSurfaceExposureFraction()
+  {
+    bool result = setString(OS_AirflowNetworkDuctViewFactorsFields::DuctSurfaceExposureFraction, "");
+    OS_ASSERT(result);
+  }
+
+  bool AirflowNetworkDuctViewFactors_Impl::setDuctSurfaceEmittance(double ductSurfaceEmittance)
+  {
+    bool result = setDouble(OS_AirflowNetworkDuctViewFactorsFields::DuctSurfaceEmittance, ductSurfaceEmittance);
+    return result;
+  }
+
+  void AirflowNetworkDuctViewFactors_Impl::resetDuctSurfaceEmittance()
+  {
+    bool result = setString(OS_AirflowNetworkDuctViewFactorsFields::DuctSurfaceEmittance, "");
+    OS_ASSERT(result);
+  }
+
+  boost::optional<AirflowNetworkLinkage> AirflowNetworkDuctViewFactors_Impl::optionalLinkage() const
+  {
+    return getObject<ModelObject>().getModelObjectTarget<AirflowNetworkLinkage>(OS_AirflowNetworkDuctViewFactorsFields::LinkageName);
+  }
+
+  boost::optional<double> AirflowNetworkDuctViewFactors_Impl::getViewFactor(const PlanarSurface &surf) const
+  {
+    for (const ModelExtensibleGroup& group : castVector<ModelExtensibleGroup>(extensibleGroups())) {
+      OptionalString name = group.getString(0);
+      if (name) {
+        if (name.get() == surf.name()) {
+          OptionalDouble value = group.getDouble(1);
+          if (value) {
+            return value;
+          } else {
+            LOG(Error, "Could not read surface view factor " << group.groupIndex() << " in " << briefDescription() << ".");
+            return boost::none;
+          }
+        }
+      } else {
+        LOG(Error, "Could not read surface name " << group.groupIndex() << " in " << briefDescription() << ".");
+      }
+    }
+    return boost::none;
+  }
+
+  bool AirflowNetworkDuctViewFactors_Impl::setViewFactor(const PlanarSurface &surf, double F)
+  {
+    //std::vector<unsigned> ids = castVector<WorkspaceExtensibleGroup>(extensibleGroups())  getSourceFieldIndices(surf.handle());
+    std::vector<unsigned> ids = getSourceIndices(surf.handle());
```

Comment:
Did you add unit tests for this?  I can't tell if it would work just by looking at it.